src/source/get.js

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2025 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { Response } from '@adobe/fetch';
+import { HelixStorage } from '@adobe/helix-shared-storage';
+import { createErrorResponse } from '../contentbus/utils.js';
+
+/**
+ * Get the headers for the response.
+ *
+ * @param {*} meta The metadata that contains many of the headers
+ * @param {number} length The content length
+ * @param {string} id The ID to store in the X-da-id header
+ * @return {Object} headers
+ */
+function getHeaders(meta, length, id) {
+  const headers = {
+    'Content-Type': meta.ContentType,
+    'Last-Modified': meta.LastModified.toUTCString(),
+    'X-da-id': id,
+  };
+  if (length) {
+    headers['Content-Length'] = length;
+  }
+  if (meta.ETag) {
+    headers.ETag = meta.ETag;
+  }
+  return headers;
+}
+
+async function accessSource(context, info, headRequest) {
+  const { log } = context;
+
+  const storage = HelixStorage.fromContext(context);
+  const bucket = storage.sourceBus();
+
+  const { org, site, resourcePath: key } = info;
+  const path = `${org}/${site}${key}`;
+
+  try {
+    if (headRequest) {
+      const head = await bucket.head(path);
+      if (!head) {
+        return new Response('', { status: 404 });
+      } else {
+        const headers = getHeaders(head, head.ContentLength, head.Metadata.id);
+        return new Response('', { status: head.$metadata.httpStatusCode, headers });
+      }
```

Comment:
I think this is just a stylistic thing: I usually don't make an `else` statement, if the  `if` branch already ends with a `return`, but either way is fine for me 🙂